### ISSUE TITLE
[Feature Request] `juliaup shell` to temporarily activate version

### ISSUE THREAD

**Issue by goerz** [JuliaLang/juliaup#605]
It would be great to have, e.g., `juliaup shell rc` to *temporarily* (for the lifetime of the current shell) activate a particular version of Julia, as opposed to `juliaup default rc` which activates the version system-wide. I'm borrowing the `shell` sub-command from `pyenv` here. Any other name or other way of achieving this would be fine.

**Comment by uncomfyhalomacro** [comment]
This is probably a small code change that anyone can do. I could try this out maybe after a week. I have exams to observe this week :smile:

**Comment by davidanthoff** [comment]
I really like the idea, _but_ I'm not so sure how easy this is if we really want to scope it to the shell from which `juliaup` was launched. The main question is where do we store that state? We would probably have to find some way to change an env var in a parent process, right?

**Comment by StefanKarpinski** [comment]
Doesn't the environment variable do exactly this?

**Comment by davidanthoff** [comment]
Yep, it probably entirely solves that. @goerz take a look at https://github.com/JuliaLang/juliaup/pull/632, that should do it, right?

I guess the only thing one could consider is whether `juliaup shell rc` should work as another way to modify `JULIAUP_CHANNEL`. But I think I'm not really convinced that it makes enough of a difference to go there.

**Comment by uncomfyhalomacro** [comment]
I think this reminds me of how `cargo` does it.

`cargo` has a `+toolchain` option which just signifies which version of rust you are going to use and run.

Not sure if that is what they want. But I think this is what I had in mind:

**Running a specific Julia version**

```bash
# latest stable release
juliaup run release

# latest release candidate
juliaup run rc

# specify version
juliaup run 1.6.0

# with julia's commandline args
juliaup run release -e 'println("Hello World!")'
```

**Comment by goerz** [comment]
Yes, `JULIAUP_CHANNEL` looks good! I'll wait for that feature to make it into the next release…

A `juliaup shell` command still seems like nice syntax sugar for this, but I agree it's not really critical if just setting `JULIAUP_CHANNEL` works as expected.

**Comment by goerz** [comment]
Also, I'm aware of the `julia +channel` feature, but I don't find it particularly useful, as I tend to use `make` to run tests / build docs / start a development REPL. The `julia` command in this case is hard-coded in the `Makefile` or the various workflow scripts. So, I'd definitely want an easy way to set just `julia` to a specific version. `JULIAUP_CHANNEL=rc make test` should definitely do the trick, though.

**Comment by davidanthoff** [comment]
@goerz also take a look at https://github.com/JuliaLang/juliaup/pull/633, that might also help.
